pysatMissions/instruments/missions_sgp4.py

```diff
@@ -100,71 +133,76 @@ def load(fnames, tag=None, sat_id=None, obs_long=0., obs_lat=0., obs_alt=0.,
     if TLE2 is not None:
         line2 = TLE2
 
-    # create satellite from TLEs and assuming a gravity model
+    if num_samples is None:
+        num_samples = 100
+
+    # Create satellite from TLEs and assuming a gravity model
     # according to module webpage, wgs72 is common
     satellite = twoline2rv(line1, line2, wgs72)
 
-    # Extract list of times from filenames and sat_id
-    times = mcore._get_times(fnames, sat_id)
+    # Extract list of times from filenames and inst_id
+    times, index, dates = ps_meth.generate_times(fnames, num_samples,
+                                                 freq=cadence)
 
-    # create list to hold satellite position, velocity
+    # Create list to hold satellite position, velocity
     position = []
     velocity = []
-    for timestep in times:
+    for timestep in index:
         # orbit propagator - computes x,y,z position and velocity
         pos, vel = satellite.propagate(timestep.year, timestep.month,
                                        timestep.day, timestep.hour,
                                        timestep.minute, timestep.second)
         position.extend(pos)
         velocity.extend(vel)
 
-    # put data into DataFrame
+    # Put data into DataFrame
     data = pds.DataFrame({'position_eci_x': position[::3],
                           'position_eci_y': position[1::3],
                           'position_eci_z': position[2::3],
                           'velocity_eci_x': velocity[::3],
                           'velocity_eci_y': velocity[1::3],
                           'velocity_eci_z': velocity[2::3]},
-                         index=times)
+                         index=index)
     data.index.name = 'Epoch'
 
     # TODO: add call for GEI/ECEF translation here
 
     return data, meta.copy()
```

Comment:
Note for the future.  Since you're using 'copy' here, you are probably not saving anything by moving the definition down below.  We should do a time check on this.

Reply:
added to #55